src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -650,6 +654,113 @@ def process_declarations(self, parent, nodes, arg_list):
                     "Could not process '{0}'. Statement Function declarations "
                     "are not supported.".format(str(stmtfn)))
 
+
+    @staticmethod
+    def _process_kind_selector(symbol_table, type_spec):
+        '''
+        Processes the fparser2 parse tree of the type specification of a
+        variable declaration in order to extract KIND information. This
+        information is used to determine the precision of the variable (as
+        supplied to the Symbol constructor).
+
+        :param symbol_table: the SymbolTable associated with the code \
+                             being processed.
+        :type symbol_table: :py:class:`psyclone.psyGen.SymbolTable`
+        :param type_spec: the fparser2 parse tree of the type specification.
+        :type type_spec: :py:class:`fparser.two.Fortran2003.Intrinsic_Type_Spec
+
+        :returns: the precision associated with the type specification.
+        :rtype: int or :py:class:`psyclone.psyGen.Symbol.Precision` or \
+                :py:class:`psyclone.psyGen.PrecisionSymbol` or NoneType
+
```

Comment:
NotImplementedErrors should be specified in the docstring.

Reply:
Oops.